### PR TITLE
KFSPTS-19988 adding 1099 form type to labor object code to 1099 box mapping

### PR DESCRIPTION
**Pull request by jhulslander**
Please note there is a related non-prod-sql PR that should be merged with this.

### CODE REVIEW

File: src/main/resources/edu/cornell/kfs/tax/cu-ojb-tax.xml [hunk 1]
```diff
@@ -8,6 +8,7 @@
         <field-descriptor name="versionNumber" column="VER_NBR" jdbc-type="BIGINT" locking="true" />
         <field-descriptor name="active" column="ACTV_IND" jdbc-type="VARCHAR" conversion="org.kuali.kfs.core.framework.persistence.ojb.conversion.OjbCharBooleanConversion" />
         <field-descriptor name="lastUpdatedTimestamp" column="LAST_UPDT_TS" jdbc-type="TIMESTAMP" index="true"/>
+        <field-descriptor name="formType" column="FORM_TYPE" jdbc-type="VARCHAR" />
```
Comment (cah292) [inline src/main/resources/edu/cornell/kfs/tax/cu-ojb-tax.xml:11]: Please see my note on the related SQL PR. If the "FORM_TYPE" does become part of the primary key, then either this user story or the follow-up user story should update the OJB metadata to configure it as such.

File: src/test/java/edu/cornell/kfs/tax/FormTypes1099Test.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
+package edu.cornell.kfs.tax;
+import static org.junit.jupiter.api.Assertions.*;
+
+import org.apache.commons.lang3.StringUtils;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import com.mchange.util.AssertException;
```
Comment (cah292) [inline src/test/java/edu/cornell/kfs/tax/FormTypes1099Test.java:9]: Can this import be removed?

File: src/test/java/edu/cornell/kfs/tax/FormTypes1099Test.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+	@AfterEach
+	void tearDown() throws Exception {
+	}
+
+	@Test
+	void findFormTypes1099FromFormCodeMisc() {
+		FormTypes1099 expectedType = FormTypes1099.MISC;
+		FormTypes1099 actualType = FormTypes1099.findFormTypes1099FromFormCode(FormTypes1099.MISC.formCode);
+		assertEquals(expectedType.formDescription, actualType.formDescription);
```
Comment (cah292) [inline src/test/java/edu/cornell/kfs/tax/FormTypes1099Test.java:25]: In this test method and the one below it, I would suggest including an assertion that also checks the formCode field.

File: src/test/java/edu/cornell/kfs/tax/FormTypes1099Test.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+		try {
+			FormTypes1099.findFormTypes1099FromFormCode(null);
+			fail("Should have an illegal argument exception");
+		} catch (Exception e) {
+			assertEquals(IllegalArgumentException.class, e.getClass());
+		}
+	}
+	
+	@Test
+	void findFormTypes1099FromFormCodeEmptyString() {
```
Comment (cah292) [inline src/test/java/edu/cornell/kfs/tax/FormTypes1099Test.java:46]: You may also want a test case for a search value that is non-blank but invalid.

File: src/main/java/edu/cornell/kfs/tax/FormTypes1099.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+	MISC("MISC", "1099-Misc"),
+	NEC("NEC", "1099-NEC");
+	
+	
+	public final String formCode;
+	public final String formDescription;
+	
+	private FormTypes1099 (String formCode, String formDescription) {
+		this.formCode = formCode;
+		this.formDescription = formDescription;
```
Comment (cah292) [inline src/main/java/edu/cornell/kfs/tax/FormTypes1099.java:17]: I'm noticing a lot of tab characters in the new and modified code sections of this PR's files. Can these tabs be converted into spaces instead?

**Review by cah292 — CHANGES_REQUESTED** [review]
I had a few suggestions for improvement, and one note about a change that may need to be added in the follow-up user story. Other than that, the changes look good.

File: src/main/resources/edu/cornell/kfs/tax/cu-ojb-tax.xml (hunk 1 above)
Comment (jhulslander) [reply src/main/resources/edu/cornell/kfs/tax/cu-ojb-tax.xml:11]: I have an open question to the functionals on this

File: src/test/java/edu/cornell/kfs/tax/FormTypes1099Test.java (hunk 2 above)
Comment (jhulslander) [reply src/test/java/edu/cornell/kfs/tax/FormTypes1099Test.java:9]: yes it should

File: src/test/java/edu/cornell/kfs/tax/FormTypes1099Test.java (hunk 3 above)
Comment (jhulslander) [reply src/test/java/edu/cornell/kfs/tax/FormTypes1099Test.java:25]: OK

File: src/test/java/edu/cornell/kfs/tax/FormTypes1099Test.java (hunk 4 above)
Comment (jhulslander) [reply src/test/java/edu/cornell/kfs/tax/FormTypes1099Test.java:46]: ok

File: src/main/java/edu/cornell/kfs/tax/FormTypes1099.java (hunk 5 above)
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/tax/FormTypes1099.java:17]: ok

**Comment by jhulslander** [comment]
I've made the changes suggested, there is an open question about object code / payment reason code mapping to more than one form type.  We should get an answer to that before merging.

File: src/main/java/edu/cornell/kfs/tax/FormTypes1099.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  public final String formDescription;
+
+  private FormTypes1099(String formCode, String formDescription) {
+    this.formCode = formCode;
+    this.formDescription = formDescription;
+  }
+
+  public static FormTypes1099 findFormTypes1099FromFormCode(String formCode) {
+    for (FormTypes1099 type : FormTypes1099.values()) {
+      if (StringUtils.equalsIgnoreCase(type.formCode, formCode)) {
```
Comment (cah292) [inline src/main/java/edu/cornell/kfs/tax/FormTypes1099.java:20]: Our cu-kfs Java code typically uses 4 spaces per indentation level. Can you please adjust the indentation of this PR's code for consistency with that? Thanks!

File: src/test/java/edu/cornell/kfs/tax/FormTypes1099Test.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    try {
+      FormTypes1099.findFormTypes1099FromFormCode(StringUtils.EMPTY);
+      fail(SHOULD_HAVE_AN_ILLEGAL_ARGUMENT_EXCEPTION);
+    } catch (Exception e) {
+      assertEquals(IllegalArgumentException.class, e.getClass());
+    }
+  }
+  
+  @Test
+  void findFormTypes1099FromFormCodeEmptyDummyText() {
```
Comment (cah292) [inline src/test/java/edu/cornell/kfs/tax/FormTypes1099Test.java:50]: I would suggest removing the word "Empty" from the method name, to more accurately describe the search input that is being tested.

**Review by cah292 — CHANGES_REQUESTED** [review]
I found a couple minor naming and indentation issues, but otherwise the changes look good. I will continue to wait for the functional feedback on whether any primary key adjustments are needed.

File: src/main/java/edu/cornell/kfs/tax/FormTypes1099.java (hunk 6 above)
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/tax/FormTypes1099.java:20]: ok

File: src/test/java/edu/cornell/kfs/tax/FormTypes1099Test.java (hunk 7 above)
Comment (jhulslander) [reply src/test/java/edu/cornell/kfs/tax/FormTypes1099Test.java:50]: ok

**Comment by jhulslander** [comment]
OK, I spoke with Cathy Salino for right now, we only want an object code and payment reason code mapped to only one box on one form
So I don't think the primary key needs to be adjusted.  This is ready for review and can be merged when you think it is ready

File: src/main/resources/edu/cornell/kfs/tax/cu-ojb-tax.xml (hunk 1 above)
Comment (cah292) [reply src/main/resources/edu/cornell/kfs/tax/cu-ojb-tax.xml:11]: Thanks for confirming this with the functionals. Based on the feedback you provided, I'm fine with the form type as-is.

**Review by cah292 — APPROVED** [review]
Changes look good, unit tests pass, and the related SQL PR is also ready. Merging.

**Pull request merged**